launcher/src/test/java/org/apache/spark/launcher/SparkSubmitCommandBuilderSuite.java

```diff
@@ -250,6 +250,34 @@ public void testMissingAppResource() {
     new SparkSubmitCommandBuilder().buildSparkSubmitArgs();
   }
 
+  @Test
+  public void testIsClientMode() {
+    // Default master is "local[*]"
+    SparkSubmitCommandBuilder launcher =
+            newCommandBuilder(Collections.emptyList());
+    assertTrue("By default application run in local mode",
+            launcher.isClientMode(Collections.EMPTY_MAP));
+    // --master yarn or it can be any RM
+    List<String> sparkSubmitArgs = Arrays.asList(
+            parser.MASTER,
+            "yarn");
+    launcher = newCommandBuilder(sparkSubmitArgs);
+    assertTrue("By default deploy mode is client",
+            launcher.isClientMode(Collections.EMPTY_MAP));
```

Comment:
`assertTrue("By default deploy mode is client", builder.isClientMode(Collections.EMPTY_MAP));`